addons/hr_skills/models/hr_employee_skill.py

```diff
@@ -97,13 +104,85 @@ def _create_logs(self):
         if skill_to_create_vals:
             self.env['hr.employee.skill.log'].create(skill_to_create_vals)
 
+    def _prepare_create_vals(self, vals_list):
+        vals_to_return = []
+        for vals in vals_list:
+            employee_id = vals.get('employee_id', False)
+            skill_id = vals.get('skill_id', False)
+            skill_type_id = vals.get('skill_type_id', False)
+            skill_level_id = vals.get('skill_level_id', False)
+
+            employee_skill_already_exist = self.env['hr.employee.skill'].search([
+                ('employee_id', '=', employee_id),
+                ('skill_id', '=', skill_id),
+                ('display_warning', '=', False),
+            ])
+            if employee_skill_already_exist:
+                skill_type = self.env['hr.skill.type'].browse(skill_type_id)
+                if not skill_type.is_certification:
+                    # Only one employee skill per skill no certificate
+                    employee_skill_already_exist[0].with_context(without_check=True).write({'skill_level_id': skill_level_id})
+                else:
+                    start_date = vals.get('start_date', False)
+                    stop_date = vals.get('stop_date', False)
+                    if not any(
+                        employee_skill.start_date == start_date and employee_skill.stop_date == stop_date
+                        for employee_skill in employee_skill_already_exist):
```

Comment:
Maybe this is a bit clearer?
```suggestion
                    exact_match = employee_skill_already_exist.filtered(lambda es: es.start_date ==  start_date and es.stop_date == stop_date)
                    if not exact_match:
                        vals_to_return.append(vals)
```